text/0131-improve-span-status-api.md

```diff
@@ -0,0 +1,238 @@
+   to see status information annotated in the trace, ideally with progressive
+   levels of detail as the focus of investigation narrows. The viewed status
+   information should be easy to correlate back to the domain that generated it
+   and diagnostically useful.
+
+2. **Searching/Filtering** - Developers suspect a particular status condition
+   might be occurring due to customer feedback, some behavior they observed
+   locally, in another trace, or perhaps code review. They want to search
+   collected telemetry to determine if and how often that status condition
+   occurs. If it occurs they want to explore example traces producing it to
+   better understand when it manifests and how it impacts their system. The
+   search terms should be intuitive given the developers initial knowledge
+   about the status condition they are searching for and the domain it arose from.
+
+3. **Grouping and metrics** - Developers monitoring a service want to
+   understand what kinds of status conditions are occurring and how frequently.
+   To do this they want the monitoring tool to provide UI that buckets results
+   into categories and show the top categories ranked by count of occurrence.
+   They may also want metrics that track category counts over time to identify
+   trends and deviations from the trend. Many useful groupings are based on
+   sharing a common problem symptom or common root cause, but other more coarse
+   groupings may be useful in trend analysis. At the most coarse spans can be
+   divided into some definition of "successful" and "failed" but there is no
+   consensus on how some status results should be bucketed (for example http
+   4xx results).
+
+### Scope
+
+I hope to define only the tracing API here, not an SDK API nor a wire
+protocol. I assume that a simple SDK API will follow relatively automatically
+if we first have agreement on the tracing API and OpenTelemetry can still be
+successful in an initial release without having standardized a wire protocol.
+
+The operation represented by a span might be composed of sub-operations (for
+example a function that calls into other lower level functions) and each of
+those sub-operations might have a notion of status. If those sub-operations
+aren't represented with a child span then it is out of scope how their status is
+captured.
+
+## Explanation
+
+#### Data representation of Span Status
+
+We should aim for representations that require minimal-to-no end-developer
+translation effort from the native representation and capture at least the key
+numeric/string data developers are likely to search for or relate to. I suggest
+status is this set of information:
+
+1. Domain - The name of the domain the status data applies to such as "HTTP",
+   "gRPC", "HRESULT", "POSIX". The list is end-user-extensible but common status
+   type names should be standardized.
+   (Perhaps there is already some standardization we could borrow?)
```

Comment:
Google did the same thing, and at the end they decided to remove this and support only what they called "canonical status" because custom translation were hard to keep track of.

One thing that was very important was the notion of "IsOk" on the Status itself, every domain has it's own codes, but in a lot of time was important to know if it is ok or not :)